src/RandomForestBase.js

```diff
@@ -0,0 +1,132 @@
+import {
+    DecisionTreeClassifier as DTClassfier,
+    DecisionTreeRegression as DTRegression
+} from 'ml-cart';
+import * as Utils from './utils';
+import Matrix from 'ml-matrix';
+
+/**
+ * @class RandomForestBase
+ */
+export default class RandomForestBase {
+
+    /**
+     * Create a new base random forest for a classifier or regression model.
+     * @constructor
+     * @param {object} options
+     * @param {number|String} [options.maxFeatures] - the number of features used on each estimator.
+     *        * if is an integer it selects maxFeatures elements over the sample features.
+     *        * if is a float between (0, 1), it takes the percentage of features.
+     * @param {boolean} [options.replacement] - use replacement over the sample features.
+     * @param {number} [options.seed] - seed for feature and samples selection, must be a 32-bit integer.
+     * @param {number} [options.nEstimators] - number of estimator to use.
+     * @param {object} [options.treeOptions] - options for the tree classifier, see [ml-cart]{@link https://mljs.github.io/decision-tree-cart/}
+     * @param {boolean} [options.classifier] - boolean to check if is a classifier or regression model (used by subclasses).
+     * @param {object} model - for load purposes.
+     */
+    constructor(options, model) {
+        if (options === true) {
+            this.options = model.options;
+            this.n = model.n;
+            this.indexes = model.indexes;
+
+            var Estimator = this.options.classifier ? DTClassfier : DTRegression;
+            this.estimators = model.estimators.map(est => Estimator.load(est));
+        } else {
+            this.options = options;
+        }
+    }
+
+    /**
+     * Train the decision tree with the given training set and labels.
+     * @param {Matrix|Array} trainingSet
+     * @param {Array} trainingValues
+     */
+    train(trainingSet, trainingValues) {
+        trainingSet = Matrix.checkMatrix(trainingSet);
+
+        this.options.maxFeatures = this.options.maxFeatures || trainingSet.columns;
```

Comment:
I think it's generally better to extract the values from the options and then use local variables instead of always reading the `this.options.value` and modifying those values.